rasa_sdk/forms.py

```diff
@@ -705,7 +601,31 @@ async def run(
         # Validation events include events for extracted slots
         return validation_events
 
-    async def extract_custom_slots(
+    async def required_slots(
```

Comment:
Why does this method have to be `async`? Also can we not remove it  and use `domain_slots` directly? Or is it preserved to continue for the user to override it as in 2.0?

Reply:
Yep the idea was to not change the signature of this function to make the migration easier. If we change it from `async` to `non-async` then the users would need to introduce loops in their custom implementations of `required_slots` when migrating from 2.0 to 3.0